mapbox.d.ts

```diff
@@ -1,202 +1,215 @@
-
-    export function setZoomLevel(arg: SetZoomLevelOptions): Promise<any>;
-    export function getZoomLevel(): Promise<number>;
-
-    export function setTilt(arg: SetTiltOptions): Promise<any>;
-    export function getTilt(): Promise<number>;
-
-    export function addPolygon(arg: AddPolygonOptions): Promise<any>;
-
-    export function animateCamera(arg: AnimateCameraOptions): Promise<any>;
-
-    export function requestFineLocationPermission(): Promise<any>;
-    export function hasFineLocationPermission(): Promise<boolean>;
-
-    export function getViewport(): Promise<Viewport>;
-    export function downloadOfflineRegion(arg: DownloadOfflineRegionOptions): Promise<any>;
-    export function listOfflineRegions(arg: ListOfflineRegionsOptions): Promise<Array<OfflineRegion>>;
-    export function deleteOfflineRegion(arg: DeleteOfflineRegionOptions): Promise<any>;
+     * Set this in case you want to later pass it to 'removeMarker'.
+     */
+    id?: any;
+    title?: string;
+    subtitle?: string;
+  }
+
+  export interface SetZoomLevelOptions {
+    level: number;
+    animated: boolean;
+  }
+
+  export interface SetTiltOptions {
+    /**
+     * default 30 (degrees)
+     */
+    pitch: number;
+    /**
+     * default 5000 (milliseconds)
+     */
+    duration: number;
+  }
+
+  export interface ShowOptionsMargins {
+    left?: number;
+    right?: number;
+    top?: number;
+    bottom?: number;
+  }
+
+  export interface Bounds {
+    north: number;
+    east: number;
+    south: number;
+    west: number;
+  }
+
+  export interface Viewport {
+    bounds: Bounds;
+    zoomLevel: number;
+  }
+
+  export interface DeleteOfflineRegionOptions {
+    /**
+     * The name of the offline region to delete.
+     */
+    name: string;
+  }
+
+  export interface ListOfflineRegionsOptions {
+    /**
+     * On Android you need to pass this in in case you haven't done so in a previous function (like 'show')
+     */
+    accessToken?: string;
+  }
+
+  export interface OfflineRegion {
+    name: string;
+    bounds: Bounds;
+    minZoom: number;
+    maxZoom: number;
+    style: MapStyle;
+  }
+
+  export interface DownloadProgress {
+    name: string;
+    completed: number;
+    expected: number;
+    percentage: number;
+    complete: boolean;
+    /**
+     * Android only, the size in bytes of the download so far.
+     */
+    completedSize?: number;
+  }
+
+  export interface DownloadOfflineRegionOptions extends OfflineRegion {
+    /**
+     * On Android you need to pass this in in case you haven't done so in a previous function (like 'show')
+     */
+    accessToken?: string;
+    onProgress: (data: DownloadProgress) => void;
+  }
+
+  /**
+   * The options object passed into the show function.
+   */
+  export interface ShowOptions {
+    accessToken: string;
+    /**
+     * default 'streets'
+     */
+    style?: MapStyle;
+    margins?: ShowOptionsMargins;
+    center?: LatLng;
+    markers?: Array<Markers>;
```

Comment:
Yes it may miss a few props, please hack your way around. Oh, heads-up: there's work in progress for converting the plugin to TS as well as adding an option to specify the map in XML (instead of creating it in JS/TS). So you might want to hold off just a little if you need those things. Or at least just "use" the plugin and not touch its code too much at this point ;)